File: mirea/cources/2026-2027/aii-july/05/Part-04-Sorted-two-pointers-and-checklist/examples/example-03.py
# Пример: количество пар с суммой не больше s двумя указателями после сортировки.
# Ввод: число тестов, затем для каждого теста n, s и массив a.
# Вывод: для каждого теста число пар (i, j), i < j, таких что a[i] + a[j] <= s.

def solve():
    n, s = map(int, input().split())
    a = list(map(int, input().split()))

    a.sort()

    left = 0
    right = n - 1
    answer = 0
    while left < right:
        current_sum = a[left] + a[right]
        if current_sum <= s:
            # Для фиксированного left подходят все вторые индексы от left + 1 до right.
            answer += right - left
            left += 1
        else:
            # Правый элемент слишком велик, уменьшаем сумму.
            right -= 1

    print(answer)


tests = int(input())
for _ in range(tests):
    solve()

File: mirea/cources/2026-2027/aii-july/05/Part-04-Sorted-two-pointers-and-checklist/examples/example-03.cpp
// Пример: количество пар с суммой не больше s двумя указателями после сортировки.
// Ввод: число тестов, затем для каждого теста n, s и массив a.
// Вывод: для каждого теста число пар (i, j), i < j, таких что a[i] + a[j] <= s.

#include <algorithm>
#include <iostream>
#include <vector>

using namespace std;

void solve() {
    int n;
    long long s;
    cin >> n >> s;
    vector<long long> a(n);
    for (int i = 0; i < n; i++) {
        cin >> a[i];
    }

    sort(a.begin(), a.end());

    int left = 0;
    int right = n - 1;
    long long answer = 0;
    while (left < right) {
        long long sum = a[left] + a[right];
        if (sum <= s) {
            // Для фиксированного left подходят все вторые индексы от left + 1 до right.
            answer += right - left;
            left++;
        } else {
            // Правый элемент слишком велик, уменьшаем сумму.
            right--;
        }
    }

    cout << answer << '\n';
}

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int tests;
    cin >> tests;
    while (tests-- > 0) {
        solve();
    }

    return 0;
}